### PR TITLE
Prevent infinite loop when recursive required field is missing

### PR DESCRIPTION
**Pull request by Bachmann1234**
Got bit by this one when testing my error flows

When trying to get all the errors in infinitely recurses and explodes

### CODE REVIEW

File: marshmallow/fields.py [hunk 1]
```diff
@@ -440,6 +441,8 @@ def _validate_missing(self, value):
         `value` should be considered missing.
         """
         if value is missing_ and hasattr(self, 'required'):
+            if self.nested == _RECURSIVE_NESTED:
+                raise ValidationError(['Missing data for required field.'])
```
Comment (sloria) [inline marshmallow/fields.py]: Use `self.fail('required')` here, so that the error message can be configured.

Comment (Bachmann1234) [inline marshmallow/fields.py]: Oh, did not notice that. Fixed!

**Comment by sloria** [comment]
Thanks!

**Comment by Bachmann1234** [comment]
w00t! No problem

**Pull request merged**